app/models/type_de_champ.rb

```diff
@@ -298,9 +297,18 @@ def self.type_de_champ_types_for(procedure, user)
     .merge(include: { types_de_champ: TYPES_DE_CHAMP_BASE })
 
   def self.as_json_for_editor
-    includes(piece_justificative_template_attachment: :blob,
-      types_de_champ: [piece_justificative_template_attachment: :blob])
-      .as_json(TYPES_DE_CHAMP)
+    includes(piece_justificative_template_attachment: :blob, types_de_champ: [piece_justificative_template_attachment: :blob]).as_json(TYPES_DE_CHAMP).map do |hash|
+      hash[:id] = hash.delete('stable_id')
+
+      if hash[:types_de_champ].present?
+        hash[:types_de_champ] = hash[:types_de_champ].map do |hash|
+          hash[:id] = hash.delete('stable_id')
```

Comment:
Je crois que j'ai du mal à comprendre la tambouille qu'on fait ici et là sur les stable-id. Je me demande comment on pourrait clarifier (ou même carrément éviter) ça. Peut-être que c'est le nom qui n'est pas bon ? Ou notre modèle de données ?

Reply:
on veut exposer vers le monde extérieur `stable_id` comme `id`. Il y a un peu de tambouilles dans cette PR pour faire la transition.

Reply:
Si c'est juste pour la transition, ça me va en l'état 👍 

Cela dit si ça à vocation à rester, je veux bien une ligne de commentaire dans la méthode pour dire qqp comme `# Expose stable-ids as ids to the editor`

Reply:
Je vais déjà faire passer la transition et je ferai du nettoyage une fois le code temporaire dégagé

Reply:
j'ai implémenté `read_attribute_for_serialization` à la place de la tambouille. C'est beaucoup plus lisible.

Reply:
Trop bien, c'est clair que c'est carrément élégant avec `read_attribute_for_serialization`. Merci ! ❤️